gh-135401: Test AWS-LC as a cryptography library in CI (#135402)
python/cpython: Misc/NEWS.d/next/Tests/2025-06-11-16-52-49.gh-issue-135401.ccMXmL.rst

> Add a new GitHub CI job to test python's ssl module with AWS-LC as the backing cryptography and TLS library.

Reviewer: Is there a better link for AWS-LC? Ideally we wouldn't directly link to the repo, but to the documentation or other landing page. I can't easily find such a link, though.
```suggestion
Add a new GitHub CI job to test the :py:mod:`ssl` module with AWS-LC__ as the backing cryptography and TLS library.

__ https://github.com/aws/aws-lc
```

Author: Yes, that's the best and most consistently up-to-date documentation entry point.

> Add a new GitHub CI job to test the :py:mod:`ssl` module with `AWS-LC <https://github.com/aws/aws-lc>`_ as the backing cryptography and TLS library.

Reviewer 2: ```suggestion
Add a new GitHub CI job to test the :mod:`ssl` module with `AWS-LC <https://github.com/aws/aws-lc>`_ as the backing cryptography and TLS library.
```

`py` is the default domain so there's no need to indicate it.

Add a new GitHub CI job to test the :py:mod:`ssl` module with AWS-LC__ as the backing cryptography and TLS library.

> __ https://github.com/aws/aws-lc

Reviewer 2: Use an inline link instead of an anonymous one. This can be achieved as follows IIRC (check if this is correct) :

```rst
`text <link>`_
```

Author: yep, that [looks right](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#embedded-uris-and-aliases).

> __ https://github.com/aws/aws-lc

Author: ```suggestion
Add a new GitHub CI job to test the :py:mod:`ssl` module with `AWS-LC <https://github.com/aws/aws-lc>`_ as the backing cryptography and TLS library.
```
